src/app/clusters/level-control/level-control.cpp

```diff
@@ -905,9 +904,11 @@ void emberAfOnOffClusterLevelControlEffectCallback(EndpointId endpoint, bool new
     app::DataModel::Nullable<uint8_t> resolvedLevel;
     uint8_t temporaryCurrentLevelCache;
     uint16_t currentOnOffTransitionTime;
-    uint8_t minimumLevelAllowedForTheDevice = minLevel;
     EmberAfStatus status;
 
+    EmberAfLevelControlState * state        = getState(endpoint);
+    uint8_t minimumLevelAllowedForTheDevice = state->minLevel;
```

Comment:
What if state is null?

Reply:
This callback is only called after the caller(OnOff cluster) verified that the endpoint has a level control cluster with this check `emberAfContainsServer(endpoint, LevelControl::Id)`

If it does, stateTable has an item for that endpoint

I can still add a check for null in case this callback gets used somewhere without the aforementioned check if you prefer

Reply:
> If it does, stateTable has an item for that endpoint

I'm afraid that's not true when dynamic endpoints are in use.  `stateTable` only knows about static endpoints that have the level control cluster on them.